sirepo/package_data/template/ml/scale.py.jinja

```diff
@@ -64,21 +65,22 @@ def scale_columns(values, column_types, col_type, scaler):
     return columns
 
 
-def scale_file(path, column_types, inputs_scaler, outputs_scaler):
+def scale_file(arch_mgr, data_path, column_types, inputs_scaler, outputs_scaler):
     {% if dataFile_appMode == 'classification' or dataFile_appMode == 'regression' %}
-    v = read_data_and_encode_output_column(path, column_types)
+    v = read_data_and_encode_output_column(arch_mgr, data_path, column_types)
     {% else %}
-    v = read_data(path)
+    v = read_data(arch_mgr, data_path)
     {% endif %}
     in_idx = scale_columns(v, column_types, 'input', inputs_scaler)
     out_idx = scale_columns(v, column_types, 'output', outputs_scaler)
-    os.remove(path)
+    os.remove(arch_mgr.file_path)
     np.save('{{ scaledFile }}', v)
     return v, in_idx, out_idx
 
 
 scaled, in_idx, out_idx = scale_file(
-    '{{ dataFile }}',
+    sirepo.sim_data.ml.ArchiveManager('{{ dataFile }}'),
```

Comment:
The name DataReader seems more appropriate.

Reply:
agreed [0227019](https://github.com/radiasoft/sirepo/pull/4736/commits/02270195eb3d6ebd58565db4e4c6e1a45eafc004)